### PR TITLE
VReplication: disable foreign_key_checks for bulk data cleanup

### PR DESCRIPTION
**Pull request by mattlord**
## Description

[VReplication](https://vitess.io/docs/reference/vreplication/) performs bulk data loads and cleanup (source side on `complete` and target side on `cancel`) and for the load portion it disabled [`foreign_key_checks`](https://dev.mysql.com/doc/refman/en/server-system-variables.html#sysvar_foreign_key_checks) (as do other bulk data load tools such as [`mysqldump`](https://dev.mysql.com/doc/refman/en/mysqldump.html)), however, it did NOT do this for the bulk data cleanup. This PR also disables `foreign_key_checks` for the cleanup step to ensure that we can successfully cleanup the data when the tables being dropped/renamed (rename only an option on `complete` for the original source tables) have dependent constraints.

This doesn't necessarily have to be backported to v19, but I thought we should because foreign key support exists there and this fixes an important bug related to that support AND v19 is still in RC.

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/15232

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/vreplication/fk_test.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ import (
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
 )
 
+const testWorkflowFlavor = workflowFlavorVtctl
```
Comment (rohit-nayak-ps) [inline go/test/endtoend/vreplication/fk_test.go:37]: This should be changed to `workflowFlavorRandom` and also tested locally with `workflowFlavorVtctld`

**Review by rohit-nayak-ps — APPROVED** [review]
Nice compact solution.

File: go/test/endtoend/vreplication/fk_test.go (hunk 1 above)
Comment (mattlord) [reply go/test/endtoend/vreplication/fk_test.go:37]: Tested locally with workflowFlavorVtctld and it passes:
```
❯ git diff | cat
diff --git a/go/test/endtoend/vreplication/fk_test.go b/go/test/endtoend/vreplication/fk_test.go
index b7a88aee65..b15856501a 100644
--- a/go/test/endtoend/vreplication/fk_test.go
+++ b/go/test/endtoend/vreplication/fk_test.go
@@ -34,7 +34,7 @@ import (
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
 )

-const testWorkflowFlavor = workflowFlavorVtctl
+const testWorkflowFlavor = workflowFlavorVtctld

 // TestFKWorkflow runs a MoveTables workflow with atomic copy for a db with foreign key constraints.
 // It inserts initial data, then simulates load. We insert both child rows with foreign keys and those without,

 ❯ make build
Thu Feb 15 15:12:32 EST 2024: Building source tree

❯ go test -v -timeout 15m -run ^TestFKWorkflow$ vitess.io/vitess/go/test/endtoend/vreplication
VTDATAROOT is /opt/vtdataroot/vreple2e_687716
=== RUN   TestFKWorkflow
E0215 15:13:19.528356   41743 vtorc_process.go:103] configuration - {
	"Debug": true,
	"ListenAddress": ":0",
	"MySQLTopologyUser": "orc_client_user",
	"MySQLTopologyPassword": "orc_client_user_password",
	"MySQLReplicaUser": "vt_repl",
	"MySQLReplicaPassword": "",
	"RecoveryPeriodBlockSeconds": 1
}
=== RUN   TestFKWorkflow/insertInitialFKData
=== RUN   TestFKWorkflow/vtctlclient_vdiff_fktarget.fk
=== RUN   TestFKWorkflow/vtctlclient_vdiff_fktarget.fk#01
--- PASS: TestFKWorkflow (52.09s)
    --- PASS: TestFKWorkflow/insertInitialFKData (0.01s)
    --- PASS: TestFKWorkflow/vtctlclient_vdiff_fktarget.fk (5.06s)
    --- PASS: TestFKWorkflow/vtctlclient_vdiff_fktarget.fk#01 (5.06s)
PASS
ok  	vitess.io/vitess/go/test/endtoend/vreplication	53.235s
```

 Will change it to random now and push that change.

File: go/vt/vttablet/tabletmanager/vdiff/stats_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func TestVDiffStats(t *testing.T) {
 		defer testStats.controllers[id].TableDiffPhaseTimings.Record(phase, time.Now())
 		time.Sleep(sleepTime)
 	}
-	want := int64(1.2 * float64(sleepTime)) // Allow 20% overhead for recording timing
+	want := int64(10 * float64(sleepTime)) // Allow 1000% overhead for recording timing on flaky test hosts
```
Comment (mattlord) [inline go/vt/vttablet/tabletmanager/vdiff/stats_test.go:57]: This addresses unrelated test flakes seen.

File: go/vt/vttablet/tabletmanager/vdiff/stats_test.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func TestVDiffStats(t *testing.T) {
 		defer testStats.controllers[id].TableDiffPhaseTimings.Record(phase, time.Now())
 		time.Sleep(sleepTime)
 	}
-	want := int64(1.2 * float64(sleepTime)) // Allow 20% overhead for recording timing
+	want := 10 * sleepTime // Allow 10x overhead for recording timing on flaky test hosts
```
Comment (mattlord) [inline go/vt/vttablet/tabletmanager/vdiff/stats_test.go:57]: This addresses unrelated test flakes seen.

**Review by notfelineit — APPROVED** [review]

**Review by frouioui — APPROVED** [review]

**Pull request merged**